ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp

```diff
@@ -456,47 +487,69 @@ void AirsimROSWrapper::car_state_timer_cb(const ros::TimerEvent& event)
 
         // todo this is global origin
         origin_geo_point_pub_.publish(origin_geo_point_msg_);
-        // iterate over drones
-        for (auto& fscar_ros: fscar_ros_vec_)
+        // iterate over cars
+        for (auto& fscar_ros : fscar_ros_vec_)
         {
-            // get drone state from airsim
-            std::unique_lock<std::recursive_mutex> lck(car_control_mutex_);
-            fscar_ros.curr_car_state = airsim_client_.getCarState(fscar_ros.vehicle_name);
-            lck.unlock();
+
+            // get car state from airsim
+            {
+                ros_bridge::Timer timer(&getCarStateStatistics);
+                std::unique_lock<std::recursive_mutex> lck(car_control_mutex_);
+                fscar_ros.curr_car_state = airsim_client_.getCarState(fscar_ros.vehicle_name);
+                lck.unlock();
+            }
+
             ros::Time curr_ros_time = ros::Time::now();
 
-            // convert airsim drone state to ROS msgs
+            // convert airsim car state to ROS msgs
             fscar_ros.curr_odom_ned = get_odom_msg_from_airsim_state(fscar_ros.curr_car_state);
             fscar_ros.curr_odom_ned.header.frame_id = fscar_ros.vehicle_name;
             fscar_ros.curr_odom_ned.child_frame_id = fscar_ros.odom_frame_id;
             fscar_ros.curr_odom_ned.header.stamp = curr_ros_time;
 
-            msr::airlib::GeoPoint gps_location = airsim_client_.getGpsData("Gps", fscar_ros.vehicle_name).gnss.geo_point;
-            fscar_ros.gps_sensor_msg = get_gps_sensor_msg_from_airsim_geo_point(gps_location);
+            {
+                ros_bridge::Timer timer(&getGpsDataStatistics);
+                msr::airlib::GeoPoint gps_location = airsim_client_.getGpsData("Gps", fscar_ros.vehicle_name).gnss.geo_point;
+                fscar_ros.gps_sensor_msg = get_gps_sensor_msg_from_airsim_geo_point(gps_location);
+            }
             fscar_ros.gps_sensor_msg.header.stamp = curr_ros_time;
 
-            // publish to ROS!  
-            fscar_ros.odom_local_ned_pub.publish(fscar_ros.curr_odom_ned);
+            // publish to ROS and keep track of incoming messages!
+            {
+                ros_bridge::ROSMsgCounter counter(&odom_local_ned_pub_statistics);
+                fscar_ros.odom_local_ned_pub.publish(fscar_ros.curr_odom_ned);
+            }

```

Comment:
counter (the extra scope is not needed but it adds clarity as to what command it is trying to count)